sdg/outputs/OutputGeoJson.py

```diff
@@ -0,0 +1,263 @@
+            A prefix added before the indicator id to construct a filename for
+            each geojson file.
+        exclude_columns : list
+            A list of strings, each a column name in the indicator data that
+            should not be included in the disaggregation. This is typically
+            for any columns that mirror the region referenced by the id column.
+        id_replacements : dict
+            An optional for with replacements to apply to the values in
+            the id_column. This is typically used if another column exists which
+            "mirrors" what would be in an id column, to avoid duplicate work.
+            For example, maybe a "Region" column exists with the names of the
+            regions as values. This can be used to "map" those region names to
+            geocodes, and save you the work of maintaining a separate id column.
+        """
+        if translations is None:
+            translations = []
+        if exclude_columns is None:
+            exclude_columns = []
+        if id_replacements is None:
+            id_replacements = {}
+
+        OutputBase.__init__(self, inputs, schema, output_folder, translations)
+        self.geojson_file = geojson_file
+        self.name_property = name_property
+        self.id_property = id_property
+        self.id_column = id_column
+        self.output_subfolder = output_subfolder
+        self.filename_prefix = filename_prefix
+        self.exclude_columns = exclude_columns
+        self.id_replacements = id_replacements
+        self.geometry_data = self.fetch_geometry_data()
+
+
+    def fetch_geometry_data(self):
+        """Grab the data referenced by the "geojson_file" parameter.
+
+        Returns
+        -------
+        dict
+            Parsed JSON as a Python dict.
+        """
+        file = None
+        data = None
+        if self.geojson_file.startswith('http'):
+            file = urlopen(self.geojson_file)
+            data = file.read().decode('utf-8')
+        else:
+            file = open(self.geojson_file)
+            data = file.read()
+        file.close()
+
+        data = json.loads(data)
+        return data
+
+
+    def indicator_has_geocodes(self, indicator):
+        """Determine if an indicator has geographical data.
+
+        Parameters
+        ----------
+        indicator : Indicator object
+            An Indicator object to examine.
+
+        Returns
+        -------
+        boolean
+            True if the indicator has geographical data, False otherwise.
+        """
+        # Make sure the indicator has data at all.
+        if not indicator.has_data():
+            return False
+        # Make sure the indicator has a geocode column.
+        cols = indicator.data.columns
+        if self.id_column not in cols:
+            return False
+        # Make sure that column has data in it.
+        return indicator.data[self.id_column].any()
+
+
+    def build(self, language=None):
```

Comment:
Some splitting up of this function might be required in this function to avoid nested `if`s and `for`s

Reply:
Agreed, I've tried to un-nest it a bit.